src/PHPMailer.php

```diff
@@ -3163,15 +3163,6 @@ public function encodeHeader($str, $position = 'text')
             $encoded = $this->wrapText($encoded, $maxlen, true);
             $encoded = str_replace('=' . static::$LE, "\n", trim($encoded));
             $encoded = preg_replace('/^(.*)$/m', ' =?' . $this->CharSet . "?$encoding?\\1?=", $encoded);
```

Comment:
@Synchro Can we simply use `us-ascii` here instead of `$this->CharSet` if `!$this->has8bitChars($encoded)`?

Reply:
I think so - the only hole would be if it uses some unusual 7-bit charset that was not ascii compatible.

Reply:
It would be cool to add a test to create headers in both formats.

Reply:
> I think so - the only hole would be if it uses some unusual 7-bit charset that was not ascii compatible.

Do we need to handle this case somehow, if we can handle it all?

> It would be cool to add a test to create headers in both formats.

Added in 9e1f8f1 .

Reply:
Great stuff, thanks. Anyone that's using a non-ascii-compatible charset probably has bigger problems, so I don't think we need to worry about it...